### PR TITLE
Replace web3 with ethers in deployment

### PR DESCRIPTION
**Pull request by leomassazza**
Add a wrapper object in Deployer.js to start the migration process from web3 to ethers.js.
Deployer.web3 is replaced with an intermediate object **provider** that will hold also ethers as a provider option. (Deployer.provider.web3, Deployer.provider.ethers).

In this PR also some direct internal usage of web3 or web3_utils is replaced by its ethers counterpart in deploy.js

### CODE REVIEW

File: publish/src/Deployer.js [hunk 1]
```diff
@@ -48,15 +48,23 @@ class Deployer {
 		this.ignoreSafetyChecks = ignoreSafetyChecks;
 
 		// Configure Web3 so we can sign transactions and connect to the network.
```
Comment (ajsantander) [inline publish/src/Deployer.js:50]: Nit: Perhaps update the comment here to clarify why we're setting up two different providers here?

File: publish/src/commands/deploy.js [hunk 2]
```diff
@@ -76,7 +76,7 @@ const deploy = async ({
 
 	// OVM uses a gas price of 0 (unless --gas explicitely defined).
 	if (useOvm && gasPrice === DEFAULTS.gasPrice) {
-		gasPrice = w3utils.toBN('0');
+		gasPrice = ethers.constants.Zero;
```
Comment (ajsantander) [inline publish/src/commands/deploy.js:79]: 💯

**Review by ajsantander — APPROVED** [review]
LGTM @leomassazza 

I would document the process a bit in the PR description. Why we're doing this, and what would the next steps/PRs be.

File: publish/src/commands/deploy.js [hunk 3]
```diff
@@ -311,16 +311,16 @@ const deploy = async ({
 		currentSynthetixSupply = await oldSynthetix.methods.totalSupply().call();
 
 		// inflationSupplyToDate = total supply - 100m
-		const inflationSupplyToDate = w3utils
-			.toBN(currentSynthetixSupply)
-			.sub(w3utils.toBN(w3utils.toWei((100e6).toString())));
+		const inflationSupplyToDate = ethers.utils
```
Comment (justinjmoses) [inline publish/src/commands/deploy.js:314]: [np (nitpick)] i'd probably destruct these utils at the top of the file

File: publish/src/commands/replace-synths.js [hunk 4]
```diff
@@ -132,7 +132,8 @@ const replaceSynths = async ({
 		},
 	};
 
-	const { web3, account } = deployer;
+	const { account } = deployer;
+	const web3 = deployer.provider.web3;
```
Comment (justinjmoses) [inline publish/src/commands/replace-synths.js:136]: are you sure there are no other instances like this around? would be tricky to find them using search...

**Review by justinjmoses — APPROVED** [review]
Just a nitpick and a seconding of Ale's comments

File: publish/src/commands/replace-synths.js (hunk 4 above)
Comment (leomassazza) [reply publish/src/commands/replace-synths.js:136]: I did a search not only of mentions but also checked which instances included Deployer to see if there was any direct or indirect reference. AFAIK there's no other usage.

**Pull request merged**